src/graphiql-fetcher.ts

```diff
@@ -0,0 +1,32 @@
+import { SubscriptionClient } from './client';
+
+export const graphQLFetcher = (subscriptionsClient: SubscriptionClient, fallbackFetcher: Function) => {
+  let activeSubscriptionId: number | null = null;
+
+  return (graphQLParams: any) => {
+    if (subscriptionsClient && activeSubscriptionId !== null) {
+      subscriptionsClient.unsubscribe(activeSubscriptionId);
+    }
+
+    if (subscriptionsClient && graphQLParams.query.startsWith('subscription')) {
```

Comment:
I agree with @NeoPhi. Instead think you can do pretty much what this does:

https://github.com/apollographql/graphql-anywhere/pull/44/files#diff-52bc6e1910f2b72f10d27006c255168fR55

If you don't want to add the graphql-js dependency, you can write that in the comment there and just use `any` for all the types. But you should definitely respect the operationName as well. GraphiQL is one of the few cases where it's actually useful for people to have many definitions in a document but only execute one. This GraphQL fetcher should support that.